### ISSUE TITLE
séparer un token en deux

### ISSUE THREAD

**Issue by sylvainkahane** [Arborator/arborator-frontend#183]
Dans https://arboratorgrew.elizia.net/#/projects/M1_ParisStories2023/ParisStories_2023_superTuteur_parisStories2023SuperTuteur_skahane,
phrase 10, j'ai voulu séparé tutoie-moi en deux. J'ai pu le faire en cliquant sur le token mais ensuite impossible de sauver.
<img width="1338" alt="image" src="https://user-images.githubusercontent.com/23051796/212566719-53772b96-78df-4e8f-9d1a-12ed6f72b5bc.png">
En plus, c'est un peu nul parce que ça efface tous les traits. Avant ils étaient dupliqué sur chaque token, c'était mieux. Certains traits comme AlignBegin et AlignEnd pourrait être gérés intelligemment, AlignBegin restant sur le premier token et AlignEnd sur le deuxième.

**Comment by B-CARON** [comment]
On peut sauvegarder si on met les liens syntaxiques. Mais l'effacement des
annotations est un gros problème. La solution de tout dupliquer est
intéressante.

Le dim. 15 janv. 2023, 21:58, Sylvain Kahane ***@***.***> a
écrit :

**Comment by kirianguiller** [comment]
@khansadaoudi , some more context for helping the resolution of this task :
- When a user wants to bulk split/add/remove tokens, he can do so by dragging+hovering the sentence text field
- A dialog will open, where the user can then manually split add or remove all the tokens he hovered
- In the source code, this dialog is called TokensReplaceDialog.vue (before it was named TokenDialog.vue, but I just changed the name in commit https://github.com/Arborator/arborator-frontend/commit/f2f99d13e77e28e4edcbc2f1ab9c65c072cfab3c for more clarity)

**Comment by kirianguiller** [comment]
[I fixed the issue](https://github.com/kirianguiller/conllup-js/commit/1efdc69505d7e86248bdff3a0a4820763b6d2952) (as it was dealing with some code I wrote), if a split token was the governor of others token, the HEAD of each governor would be deleted, but not the DEPREL, which then leaded to a grew error as if a HEAD exist on a token, the DEPREL need to exist too.

So at the moment, I changed the code so both the DEPREL and HEAD are deleted (the desired feature of smartly keeping deprel will be done next, with a new issue that I'm opening).

The reason why we need a new issue is that the desired behaviors (copying the features when splitting and keeping the dependency relation on the split token) can only be applied in some of the use-cases of the TokensReplace feature.
Indeed, the TokensReplace feature can do the following things : 
- delete completely one or many tokens
- add completely new tokens located after a given token
- replace one token to many tokens (probably the most frequent use-case)
- replace many tokens to one token
- replace many tokens to many tokens
- rename a token (this can (and should) be done in other way, such as directly clicking on a token and modifying the FORM feature)

But all of these 6 use cases use the same `replaceArrayOfTokens()` method (from [conllup-js](https://github.com/kirianguiller/conllup-js)) and only a few of them are compatible with the desired behaviors.

Thanks for reporting the bug.
